src/schema/v2/partnerArtworksMutation.ts

```diff
@@ -0,0 +1,136 @@
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import {
+  GravityMutationErrorType,
+  formatGravityError,
+} from "lib/gravityErrorHandler"
+import { GraphQLObjectType } from "graphql"
+import { InternalIDFields } from "./object_identification"
+import { GraphQLUnionType } from "graphql"
+interface Input {
+  id: string
+  artsyShippingDomestic: boolean | null
+  artsyShippingInternational: boolean | null
+  location: string | null
+}
+
+const PartnerArtworksType = new GraphQLObjectType<any, ResolverContext>({
+  name: "PartnerArtworks",
+  fields: () => ({
+    ...InternalIDFields,
+    success: { type: GraphQLInt },
+    errors: {
+      type: new GraphQLObjectType({
+        name: "PartnerArtworksError",
+        fields: {
+          count: { type: GraphQLInt },
+          ids: { type: GraphQLList(GraphQLString) },
+        },
+      }),
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationSuccessType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutationSuccess",
+  isTypeOf: (data) => data.success || data.errors,
+  fields: () => ({
+    partnerArtworks: {
+      type: PartnerArtworksType,
+      resolve: (partnerArtworks) => partnerArtworks,
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationFailureType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationType = new GraphQLUnionType({
+  name: "UpdatePartnerArtworksMutationType",
+  types: [
+    UpdatePartnerArtworksMutationSuccessType,
+    UpdatePartnerArtworksMutationFailureType,
+  ],
+})
+
+export const updatePartnerArtworksMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutation",
+  description: "Update all artworks that belong to the partner",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "ID of the partner",
+    },
+    artsyShippingDomestic: {
+      type: GraphQLBoolean,
+      description: "Whether Artsy domestic shipping should be enabled",
+    },
+    artsyShippingInternational: {
+      type: GraphQLBoolean,
+      description: "Whether Artsy international shipping should be enabled",
+    },
+    location: {
+      type: GraphQLString,
+      description: "The partner location ID to assign",
+    },
+  },
+  outputFields: {
+    partnerArtworksOrError: {
+      type: UpdatePartnerArtworksMutationType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: (
+    { id, artsyShippingDomestic, artsyShippingInternational, location },
+    { updatePartnerArtworksLoader }
+  ) => {
+    if (!updatePartnerArtworksMutation) {
```

Comment:
Not sure when `!updatePartnerArtworksMutation` would be true. Did you mean to check if the loader exists (which would indicate the MP request was lacking an `X-Access-Token` header if it didn't)?

Reply:
I was following [this PR](https://github.com/artsy/metaphysics/pull/3918) as this REST endpoint is only available for admins at the moment, and this seemed to match.

Reply:
I've removed this check as the admin only feature is actually already on the REST endpoint so this is redundant.

Reply:
If you notice that PR checks the _loader_ existence (not the mutation). It's still good to check the loader existence in order to fail with a valid error ('This operation only available with an access token'), vs. now where this mutation might just return `null` (w/ no error or explanation) if a user attempts it w/o a token.

MP doesn't verify tokens, but it does use the existence of them being passed as headers to decide if the authenticated loaders (and typically most mutations) have loaders made available in the context.

Reply:
Totally missed that, it all makes sense now. Thanks for the context! Just added the checking of the loader in, so now the null check is no longer needed